lib/_http_client.js

```diff
@@ -24,6 +24,18 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
+  } else if (options instanceof url.URL) {
```

Comment:
True, but that's already the case if you pass the URL as a string or the result of `url.parse`. We can think later about a way to pass a URL object along with additional options but I don't think it should block this PR.